### PR TITLE
feat: Implement missing push rules

### LINKED ISSUE #3274: Feature: Add support for Repo/Org Push rulesets
GitHub launched [push rules](https://github.blog/changelog/2024-09-10-push-rules-are-now-generally-available-and-updates-to-custom-properties/) for rulesets to GA recently where you can allow only a certain team or group to push particular paths or files. The API has documentation now to support `push` rule targets for both [Repository](https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#create-a-repository-ruleset--parameters) and [Organisation](https://docs.github.com/en/rest/orgs/rules?apiVersion=2022-11-28#create-an-organization-repository-ruleset--parameters). 

It would be great to be able to support creating and managing these programatically.

### PR DESCRIPTION
**Pull request by srgustafson8**
Adds support for the missing [Push Rules](https://github.blog/changelog/2024-09-10-push-rules-are-now-generally-available-and-updates-to-custom-properties/) - Restrict file path length, Restrict file extensions and Restrict file size with tests.

Closes #3274

### CODE REVIEW

**Comment by srgustafson8** [comment]
Tests output:

```❯ script/test.sh
testing .
?       github.com/google/go-github/v65/test/integration        [no test files]
        github.com/google/go-github/v65/test/fields             coverage: 0.0% of statements
ok      github.com/google/go-github/v65/github  8.592s  coverage: 99.4% of statements
testing example
        github.com/google/go-github/v65/example/codespaces/newusersecretwithxcrypto             coverage: 0.0% of statements
        github.com/google/go-github/v65/example/appengine               coverage: 0.0% of statements
        github.com/google/go-github/v65/example/basicauth               coverage: 0.0% of statements
        github.com/google/go-github/v65/example/codespaces/newreposecretwithxcrypto             coverage: 0.0% of statements
        github.com/google/go-github/v65/example/actionpermissions               coverage: 0.0% of statements
        github.com/google/go-github/v65/example/migrations              coverage: 0.0% of statements
        github.com/google/go-github/v65/example/commitpr                coverage: 0.0% of statements
        github.com/google/go-github/v65/example/listenvironments                coverage: 0.0% of statements
        github.com/google/go-github/v65/example/newfilewithappauth              coverage: 0.0% of statements
        github.com/google/go-github/v65/example/newrepo         coverage: 0.0% of statements
        github.com/google/go-github/v65/example/ratelimit               coverage: 0.0% of statements
        github.com/google/go-github/v65/example/newreposecretwithxcrypto                coverage: 0.0% of statements
        github.com/google/go-github/v65/example/simple          coverage: 0.0% of statements
        github.com/google/go-github/v65/example/tagprotection           coverage: 0.0% of statements
        github.com/google/go-github/v65/example/tokenauth               coverage: 0.0% of statements
        github.com/google/go-github/v65/example/topics          coverage: 0.0% of statements
testing scrape
ok      github.com/google/go-github/scrape      (cached)        coverage: 59.4% of statements
        github.com/google/go-github/scrape/example/scrape               coverage: 0.0% of statements
testing tools
ok      tools/metadata  (cached)        coverage: 81.7% of statements```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:95]: Unless there is a strong reason to use a pointer to a slice, we try very hard in this repo to avoid it.
I'm not aware of a strong need in this circumstance, but please correct me if I'm wrong.

```suggestion
	RestrictedFileExtensions []string `json:"restricted_file_extensions"`
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:93]: Question: Should this possibly be?

```
// RuleFileExtensionRestrictionParameters represents the restricted_file_extensions rule parameters.
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:100]: Unless you know otherwise, I'm thinking `int64` for `MaxFileSize` might be more useful here.

```suggestion
	MaxFileSize int64 `json:"max_file_size"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @srgustafson8 !
Just a few minor tweaks, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

If you have a moment, would you mind taking a look at some of the other outstanding PRs and give them code reviews, please? Of note is #3254 where the author is expressing desire to merge ASAP.  Thank you!

File: github/repos_rules.go
Comment (srgustafson8) [reply github/repos_rules.go:93]: Annoyingly no, the rule is called file_extension_restriction and the parameter is restricted_file_extensions

File: github/repos_rules.go
Comment (srgustafson8) [reply github/repos_rules.go:95]: I did wonder, took a look at the prior push rule addition which was a pointer. Wasn't sure if that was necessary either! will modify

**Review by gmlewis — APPROVED** [review]
Thank you, @srgustafson8 !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Review by air-hand — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin and @air-hand !
Merging.

**Pull request merged**
